pandas/tests/dtypes/test_cast.py

```diff
@@ -456,3 +457,15 @@ def test_cast_1d_arraylike_from_scalar_categorical(self):
 def test_construct_1d_ndarray_preserving_na(values, dtype, expected):
     result = construct_1d_ndarray_preserving_na(values, dtype=dtype)
     tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize('arr, dtype', [
+    (np.array(['0:0:1'], dtype='object'), 'timedelta64[ns]'),
+    (np.array(['0:0:1'], dtype='object'), 'timedelta64'),
+    (np.array(['2000'], dtype='object'), 'datetime64[ns]'),
+    (np.array(['2000'], dtype='object'), 'datetime64'),
+])
+def test_astype_nansafe(arr, dtype):
+    # GH #22100
+    result = astype_nansafe(arr, dtype)
+    is_dtype_equal(result.dtype, dtype)
```

Comment:
Should there be an assertion here?

Reply:
Piggybacking on this I think it would be better to explicitly parametrize the expected value and make assertions around that.

On a somewhat related note how do we want to handle `datetime64[ns, tz]`? Off the top of my head I can't think of how to do a conversion to that via `astype` so maybe should still raise for that?